src/app/src/components/search-page/search-page.component.ts

```diff
@@ -0,0 +1,27 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Observable } from 'rxjs';
+import { IMoviePage } from '../../models/IMoviePage.model';
+import { SearchMoviesService } from '../../services/search-movies.service';
+
+@Component({
+  selector: 'app-search-page',
+  templateUrl: './search-page.component.html',
+  styleUrls: ['./search-page.component.less'],
+})
+export class SearchPageComponent implements OnInit {
+  searchResults$ = new Observable<IMoviePage>();
+
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private searchMoviesService: SearchMoviesService
+  ) {}
+
+  ngOnInit(): void {
+    this.activatedRoute.queryParams.subscribe((result) => {
```

Comment:
you could do `const query = this.activatedRoute.snapshot.paramMap.get('query')`
and then `this.searchResults$ = this.searchMoviesService.getSearchResults(query )`

or you could also do `this.searchResults$ = this.activatedRoute.queryParams.pipe(switchMap(params => this.searchMoviesService.getSearchResults(params.query) ));`